src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.ChecksumToSolutionCache.SolutionAndInFlightCount.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
The following two types/files (ChecksumToSolutionCache and SolutionAndInFlightCount) are utility types that the main bulk of the change benefits from. I tried to keep them as clear as i coudl given that they're implementing a ref-counting system.